src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DefaultWorkspaceSemanticTokensRefreshTrigger.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.Razor;
+using Microsoft.CodeAnalysis.Razor.ProjectSystem;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer
+{
+    /// <summary>
+    /// Sends a 'workspace\semanticTokens\refresh' request each time the project changes.
+    /// </summary>
+    internal class DefaultWorkspaceSemanticTokensRefreshTrigger : ProjectSnapshotChangeTrigger
+    {
+        private readonly WorkspaceSemanticTokensRefreshPublisher _workspaceChangedPublisher;
+        private ProjectSnapshotManagerBase? _projectSnapshotManager;
+
+        internal DefaultWorkspaceSemanticTokensRefreshTrigger(WorkspaceSemanticTokensRefreshPublisher workspaceSemanticTokensRefreshPublisher)
+        {
+            _workspaceChangedPublisher = workspaceSemanticTokensRefreshPublisher;
```

Comment:
To avoid ambiguity with other workspace events lets just double down on the ambiguity 😄
```suggestion
            _publisher = workspaceSemanticTokensRefreshPublisher;
```